Added kicking of players from a running mission

FlightControl-User opened the pull request:
Through the manipulation of the user flags, players can now also be kicked.
I also added a couple of additional variables that can be modified to disable / tweak the player kicking.

DCS has a bug in the multiplayer system. Unit:destroy() does not work in MP when a player seated from a client PC is destroyed on the server.

This development however, is better, as this allows also to block slots upon preference from the mission, not just kicking. So this has become a nice set of functions to use together with missions.

I hope this change will find its way in the community.

The MOOSE framework will make heavy use of this. For example in the AIRBASEPOLICE.
Speeding pilots on the runway will be kicked!

cheers,
Sven

FlightControl-User committed:
Kicking of players mid game

FlightControl-User committed:
Added variables (removed constants)

FlightControl-User committed:
Ignore .project file

FlightControl-User committed:
Removed remaining logging

ciribob commented on SimpleSlotBlockGameGUI.lua:
```diff
@@ -169,6 +192,51 @@ function ssb.getGroupName(_slotID)
+            local _allow = ssb.shouldAllowAircraftSlot(playerID, _playerDetails.slot)
+    
+            if not _allow then
+                ssb.rejectPlayer(playerID)
```
After this - I'd clear the flag after removing the user from the slot as for the particular behaviour I assume you dont want the slot disabled forever

FlightControl-User replied:
I have deliberately not done this because then the mission is in control.

FlightControl-User replied:
Thinking about it. I could add a flag to auto unblock... I'll do that. Thanks for suggestion. Ill update.

FlightControl-User committed:
Added the changes for the slot blocker.
Now also including the automatic reopening of the slot after a player has been kicked.

FlightControl-User committed:
New project name

FlightControl-User commented:
OK. I did the following:

1. Added method ssb.allowAircraftSlot(_playerID, _slotID) to allow an aircraft slot.
2. Added method ssb.setFlagValue(_flag, _number) to set a flag value.
3. Added a control variable ssb.kickReset = true -- The slot will be automatically reset to open, after kicking the player.
4. When a player gets kicked in mid flight, the slot will be automatically re-opened.
5. Reformatted the source to 2 indents. (auto format in eclipse)
6. Improved a bit on the documentation, added an Additions section.

FlightControl-User commented:
The module is heavily tested, and will be used by the TAW competition event, which we are preparing.

FlightControl-User committed:
When a new mission has been loaded, the previous time persistent variable is reset to 0, because the model time is reset to 0 too!

FlightControl-User committed:
Alignment

FlightControl-User commented:
On mission end, i am resetting some persistent variables ... prevTimeKick.
You can publish now.

FlightControl-User commented:
Thank you!
